### DISCUSSION TITLE
Homebrew installation on ARM

### DISCUSSION THREAD

**Discussion by sosterberg** [Homebrew/discussions#357] [Getting started]
Hi,

I’m about to install Homebrew on my M1 MacBook. I’ve seen the two different versions of installation, Rosetta 2 vs. native, but what is the best way to do this?

Should I do everything via Rosetta 2 for now? And will there be a way to port those installations to native?

Or should do as much as possible native and “complement” with Rosetta 2 trying to minimize the the effort needed to transition when native is supported?

**Comment by SMillerDev** [comment]
Native isn't supported yet so that's definitely not recommended. I'd go with a rosetta install and a `rosetta-brew bundle dump && brew bundle install` when it's time to switch

**Reply by sosterberg** [reply]
Ok, got it. Thanks for clear and swift reply!

**Reply by tjluoma** [reply]
Thanks for this. I wasn't sure what the procedure would be when it _was_ time to switch but figured y'all were smart enough to figure something out and tell those of us who, well, aren't _(a category in which I include myself)_.

**Reply by numbersinmyhead** [reply]
I am thinking of installing the rosetta version too, but in future when the arm version is available, what would be the procedure for me to remove the rosetta version of homebrew?

**Comment by superlbr** [comment]
why not try https://github.com/pforret/m1_homebrew, native is so good

**Reply by SMillerDev** [reply]
For anyone reading this, this is not supported by the Hombrew project

**Reply by superlbr** [reply]
For now, brew upgrade works

**Comment by rituparnakhaund** [comment]
I need to specifically be using php@7.3 but I am facing issues while installing it . Yes I am using M1 , can someone help?

**Reply by manearaluca-amoos** [reply]
Hello, how you fixed this? I need this too.
Thank you.
